molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+        self._pool_path = os.path.expanduser(
+            os.path.join('/opt/jenkins', 'libvirt', 'images'))
+        self._sources_path = os.path.expanduser(
+            os.path.join('/opt/jenkins', 'libvirt', 'sources'))
+        for path in [self._pool_path, self._sources_path]:
+            if not os.path.exists(path):
+                os.makedirs(path, 0775)
+        self._boot_wait = 60
+
+    def _get_provider(self):
+        return 'libvirt'
+
+    def _get_platform(self):
+        return self.default_platform
+
+    @property
+    def name(self):
+        return 'libvirt'
+
+    @property
+    def instances(self):
+        return self.molecule.config.config['libvirt']['instances']
+
+    @property
+    def default_provider(self):
+        return 'libvirt'
+
+    @property
+    def default_platform(self):
+        return ''
+
+    @property
+    def provider(self):
+        return self._provider
+
+    @property
+    def platform(self):
+        return self._platform
+
+    @property
+    def host_template(self):
+        return '{} ansible_ssh_host={} ansible_ssh_private_key_file={} ansible_ssh_user={}\n'
+
+    @property
+    def valid_providers(self):
+        return [{'name': 'Libvirt'}]
+
+    @property
+    def valid_platforms(self):
+        return [{'name': 'Libvirt'}]
+
+    @property
+    def ssh_config_file(self):
+        return '.molecule/ssh_config'
+
+    @property
+    def testinfra_args(self):
+        kwargs = {
+            'ansible-inventory':
+            self.molecule.config.config['ansible']['inventory_file'],
+            'connection': 'ansible'
+        }
+
+        return kwargs
+
+    @property
+    def serverspec_args(self):
+        return dict()
+
+    @property
+    def ansible_connection_params(self):
+        params = {'connection': 'ssh'}
+
+        return params
+
+    def _define_pool(self, pool_name):
+        """
+        Define a libvirt storage pool and start it.
+
+        :param: the name of the pool
+        :param: the filesystem path for the pool
+        """
+        utilities.print_info("Creating libvirt storage pool for molecule ...")
+        pool = ET.Element('pool', type='dir')
+        ET.SubElement(pool, 'name').text = pool_name
+        target = ET.SubElement(pool, 'target')
+        ET.SubElement(target, 'path').text = self._pool_path
+        poolxml = ET.tostring(pool)
+        LOG.debug("\tXMLDesc: {}".format(poolxml))
+        newpool = self._libvirt.storagePoolDefineXML(poolxml)
+        newpool.create()
+        return newpool
+
+    def _up_pool(self, pool_name='molecule'):
+        """
+        Up or define/up a libvirt storage pool
+
+        :param: name for the pool
+        """
+        try:
+            pool_found = self._libvirt.storagePoolLookupByName(pool_name)
+            if pool_found and not pool_found.isActive():
+                pool_found.create()
+            return pool_found  # existing pool is now running
+        except libvirt.libvirtError:
```

Comment:
It's usually frown upon to have more than one line inside a try/except block.  And for good reasons, because it usually leads to unexpected exceptions being caught.  Is `self._libvirt.storagePoolLookupByName(pool_name)` the only line that can trigger a `libvirt.libvirtError:` exception or is `pool_found.create()` also likely to trigger it?  If both can trigger it, I'd at least move the return outside of the block:
```
try:
    # blah
except:
    # blih
else:
    return pool_found
```